modules/command-engine/javadsl/src/main/scala/surge/javadsl/command/AggregateRef.scala

```diff
@@ -38,31 +41,51 @@ final class AggregateRefImpl[AggId, Agg, Cmd, Event](
   private val log: Logger = LoggerFactory.getLogger(getClass)
 
   override def getState: CompletionStage[Optional[Agg]] = {
-    val status = engineStatus()
+    val result = retryWithDelay(5).flatMap { _ =>
+      val status = getEngineStatus()
 
-    val result = if (status == SurgeEngineStatus.Running) {
-      queryState
-    } else {
-      log.error(s"Engine Status: $status")
-      Future.failed(SurgeEngineNotRunningException("The engine is not running, please call .start() on the engine before interacting with it"))
+      if (status == SurgeEngineStatus.Running) {
+        queryState
+      } else {
+        log.error(s"Engine Status: $status")
+        Future.failed(
+          SurgeEngineNotRunningException("The engine is not running, please call .start()" +
+            " on the engine before interacting with it"))
+      }
     }
-
     FutureConverters.toJava(result.map(_.asJava))
   }
 
   def sendCommand(command: Cmd): CompletionStage[CommandResult[Agg]] = {
-    val status = engineStatus()
-
-    val result = if (status == SurgeEngineStatus.Running) {
-      val envelope = PersistentActor.ProcessMessage[Cmd](aggregateId.toString, command)
-      sendCommand(envelope).map(aggOpt => CommandSuccess[Agg](aggOpt.asJava)).recover { case error =>
-        CommandFailure[Agg](error)
+    val r = retryWithDelay(5).flatMap { _ =>
+      val status = getEngineStatus()
+      if (status == SurgeEngineStatus.Running) {
+        val envelope = PersistentActor.ProcessMessage[Cmd](aggregateId.toString, command)
+        sendCommand(envelope).map(aggOpt => CommandSuccess[Agg](aggOpt.asJava)).recover { case error =>
+          CommandFailure[Agg](error)
+        }
+      } else {
+        log.error(s"Engine Status: $status")
+        Future.failed(
+          SurgeEngineNotRunningException("The engine is not running, please call .start() " +
+            "on the engine before interacting with it"))
       }
-    } else {
-      log.error(s"Engine Status: $status")
-      Future.failed(SurgeEngineNotRunningException("The engine is not running, please call .start() on the engine before interacting with it"))
     }
 
-    FutureConverters.toJava(result)
+    FutureConverters.toJava(r)
   }
+
+  private def retryWithDelay(numTries: Int, delay: FiniteDuration = 50.millis): Future[Unit] = {
```

Comment:
can we rename this to "whenEngineReady"

Reply:
How about something like this : 

```
whenEngineReady[T](numTries: Int = 5, delay: FiniteDuration = 50.millis)(codeBlock: _ => Future[T]): Future[T] = {

}
```

Reply:
Done, but I kept codeblock to ` cb: => Future[T]`

Reply:
Amazing ! Thanks ;-)